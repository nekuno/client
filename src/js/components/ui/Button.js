import React, { PropTypes, Component } from 'react';
import shouldPureComponentUpdate from 'react-pure-render/function';

export default class Button extends Component {
    static propTypes = {
        text: PropTypes.string.isRequired
    };

    shouldComponentUpdate = shouldPureComponentUpdate;

    render() {
        return (
            <a href="#" className="button button-fill button-round">{this.props.text}</a>
        );
    }
}