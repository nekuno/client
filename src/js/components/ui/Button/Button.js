import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './Button.scss';

export default class Button extends Component {

    static propTypes = {
        color           : PropTypes.string,
        borderColor     : PropTypes.string,
        backgroundColor : PropTypes.string,
        disabled        : PropTypes.bool,
        size            : PropTypes.oneOf(['small']),
        textAlign       : PropTypes.string,
        onClickHandler  : PropTypes.func
    };

    handleClick() {
        const {disabled} = this.props;

        if (!disabled && this.props.onClickHandler) {
            this.props.onClickHandler();
        }
    }

    render() {
        const {textAlign, size, disabled, color, borderColor, backgroundColor, children} = this.props;

        return (
            <div className={styles.buttonWrapper + ' ' + styles[textAlign]}>
                <button className={styles.button + ' ' + styles[size]} disabled={disabled} style={{color: color, borderColor: borderColor, backgroundColor: backgroundColor}} onClick={this.handleClick.bind(this)}>
                    {children}
                </button>
            </div>
        );
    }
}