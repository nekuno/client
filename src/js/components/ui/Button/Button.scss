@import '../../../../scss/_partials/colors';

.button-wrapper {
  text-align: center;

  .button {
    background: transparent;
    border: 1px solid $white-background;
    border-radius: 30px;
    -moz-border-radius: 30px;
    -webkit-border-radius: 30px;
    color: $white-background;
    cursor: pointer;
    font-size: 16px;
    font-weight: 600;
    height: 60px;
    line-height: 60px;
    width: 70%;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    -webkit-box-flex: 1;
    -webkit-flex-shrink: 1;
    -ms-flex: 0 1 auto;
    flex-shrink: 1;
    min-width: 0;
    outline: none;

    &:disabled, &:active {
      opacity: 0.5;
    }

    &:disabled {
      cursor: default;
    }

    &.small {
      width: 180px;
      height: 40px;
      line-height: 38px;
    }
  }
  &.left {
    text-align: left;
  }

  &.center {
    text-align: center;
  }

  &.right {
    text-align: right;
  }
}