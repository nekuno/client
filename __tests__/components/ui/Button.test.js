import React from 'react';
import Button from '../../../src/js/components/ui/Button/Button.js';
import styles from '../../../src/js/components/ui/Button/Button.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test Button component', () => {
    const mockFn = jest.fn();
    const button = shallow(<Button onClickHandler={mockFn}>Lorem ipsum</Button>);

    it('should be defined', () => {
        expect(Button).toBeDefined();
    });
    it('should render correctly', () => {
        expect(button).toMatchSnapshot();
    });
    it('Button click event', () => {
        button.find(`.${styles.button}`).simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });
});
