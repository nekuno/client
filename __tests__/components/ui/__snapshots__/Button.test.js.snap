// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    onClickHandler={[MockFunction]}
  >
    Lorem ipsum
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "borderColor": undefined,
            "color": undefined,
          }
        }
      >
        Lorem ipsum
      </button>,
      "className": "buttonWrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem ipsum",
        "className": "button",
        "disabled": undefined,
        "onClick": [Function],
        "style": Object {
          "borderColor": undefined,
          "color": undefined,
        },
      },
      "ref": null,
      "rendered": "Lorem ipsum",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "borderColor": undefined,
              "color": undefined,
            }
          }
        >
          Lorem ipsum
        </button>,
        "className": "buttonWrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem ipsum",
          "className": "button",
          "disabled": undefined,
          "onClick": [Function],
          "style": Object {
            "borderColor": undefined,
            "color": undefined,
          },
        },
        "ref": null,
        "rendered": "Lorem ipsum",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
